TheFunctionChangedMyMind/README-en.md

```diff
@@ -0,0 +1,322 @@
+---
+title: The Function that changed my mind
+published: true
+description: How a single function showed me a lot of Kotlin
+tags: kotlin,functions,android,refactor
+cover_image:
+series:
+---
+
+When writting code, we should always have a place for improvement. Improvement can present itself in different ways. A proof of it is how in recent years the attention to CI (_Continuous Integration_) and CD (_Continuous Delivery_) has grown significantly. Same goes for new languages and technologies.
+
+But not all involve changing to a new language, technology or tool. Some of these improvements are simply changing how something works, adding tests, adding abstractions, changing patterns, etc. This is the case for the change I introduced on one of my projects.
+
+**Disclaimer**: If you know me in person or have been in a meetup with me presenting, there's a big chance you already heard me talking about this, even [on YouTube with the GDG Santo Domingo (in spanish)](https://www.youtube.com/watch?v=0KoXxoPHV4o&feature=youtu.be) or Kotlin/Everywhere Guadalajara and Mexico City.
+
+**Another disclaimer**: This article is not about architecture –even when it's a hot topic in the Android community– or best practices or even about how to organize an Android Application. It's just about a set of changes that showed me some Kotlin concepts. Also, I started these changes almost 3 years ago, so by the time this article is published, the libraries from Google –like `core-ktx`– will probably already have something similar or better. And just to be clear: None of the approaches discussed here are right or wrong, all the solutions depend on individual needs and individual reasons to choose them over the others.
+
+## But first… some _Context_
+
+If you're an Android Developer or have worked with Android, you probably caught the pun intended by this title. If not, let me explain it to you quickly.
+
+In Android, everything happening on your application has a `Context` and to make some stuff work, you also need a `Context`. This is just a class that Android uses to give information and access to the app.
+
+One specific case is when you want to start a new `Activity` on your app. This means –in short– launching a whole new screen or an external application. To do this, you first create an `Intent` based on your `Context` and pass the `Activity` class you want to start:
+
+```kotlin
+val myIntent = Intent(context, MyNextActivity::class.java)
+```
+
+Then you can start `MyNextActivity` from another `Activity` simply using this intent: `startActivity(myIntent)`
+
+If your application relays on activities for showing different sections/views of your application then probably you will end with a lot of these calls distributed around. In some different variants, but all of them with the same structure:
```

Comment:
//Relays is not relies, keep that in mind

If your application relies on activities for showing varied sections or views of your application, then probably you will end **UP**